src/transaction-runner.coffee

```diff
@@ -66,8 +66,22 @@ class TransactionRunner
 
         catch error
           if error instanceof chai.AssertionError
-            data.message = "Failed assertion in hooks: " + error.message
-            data.test?.status = 'fail'
+            message = "Failed assertion in hooks: " + error.message
+
+            data['results'] ?= {}
+            data['results']['general'] ?= []
+            data['results']['general'].push { severity: 'error', message: message }
```

Comment:
wait wait wait... `general` is used to report connection errors, not hooks emitted Chai Assertion errors.

I'm :-1: here for the re-use of `general` here, since it mixes up with `connection` errors together and you'll have a list of `connection error` together with possible `before` or `after` hook error?